pkgs/instrumentations/web/page-view/src/instrumentation.ts

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  InstrumentationBase,
+  isWrapped,
+} from '@opentelemetry/sandbox-instrumentation';
+import { Event } from '@opentelemetry/sandbox-api-events';
+import { EventEmitter } from '@opentelemetry/sandbox-sdk-events';
+
+import { VERSION } from './version';
+import {
+  ApplyCustomEventDataFunction,
+  PageViewInstrumentationConfig,
+} from './types';
+/**
+ * This class represents a document load plugin
+ */
+
+export class PageViewEventInstrumentation extends InstrumentationBase<unknown> {
+  static readonly instrumentationName =
+    '@opentelemetry/instrumentation-page-view';
+  readonly component: string = 'page-view-event';
+  readonly version: string = '1';
+  moduleName = this.component;
+  emitter: EventEmitter | null = null;
+  oldUrl = location.href;
+  applyCustomEventData: ApplyCustomEventDataFunction | undefined = undefined;
+
+  /**
+   *
+   * @param config
+   */
+  constructor(config: PageViewInstrumentationConfig) {
+    super(PageViewEventInstrumentation.instrumentationName, VERSION, config);
+    this.emitter = new EventEmitter(
+      PageViewEventInstrumentation.name,
+      VERSION,
+      'browser'
+    );
+    this.applyCustomEventData = config.applyCustomEventData;
+    this._patchHistoryApi();
+  }
+
+  init() {}
+
+  /**
+   * callback to be executed when page is viewed
+   */
+  private _onPageView() {
+    const pageViewEvent: Event = {
+      name: 'page_view',
+      data: {
+        url: document.documentURI as string,
+        referrer: document.referrer,
+        title: document.title,
+        type: 0,
+      },
+    };
+    this._applyCustomEventData(pageViewEvent, this.applyCustomEventData);
+    this.emitter?.emit(pageViewEvent);
```

Comment:
Doesn't the `emit" take at least 2 arguments? (It should) and then this would be

`this.emitter.emit("page_view", data)`; Where the data is the object with the attributes.

Reply:
the emit function defined in the api takes only one argument, the event (with name ,and data part of the event object)